### ISSUE TITLE
Several False Positives of stringQuotes Rule

### ISSUE THREAD

**Issue by fabb** [lesshint/lesshint#505]
**Which version of `lesshint` are you using?**
6.0.0

**How are you running `lesshint`? CLI, Node.js API, Grunt/Gulp plugin?**
Via npm run script or just `npx lesshint@6.0.0 path/to/style.less`

**What's your `.lesshintrc` configuration?**

```js
{
    "excludedFiles": ["node_modules"],
    "emptyRule": false,
    "spaceAfterPropertyName": false,
    "propertyOrdering": false,
    "zeroUnit": ["no_unit"],
    "importPath": [true, false],
    "singleLinePerSelector": false,
    "qualifyingElement": false,
    "universalSelector": false,
    "duplicateProperty": ["display"],
    "maxCharPerLine": 180,
    "importantRule": false,
    "spaceAroundComma": {"allowNewline": true},
    "spaceAfterPropertyColon": {"style": "at_least_one_space"}
}
```

**Code that's failing:**

Example 1:
```less
@import (reference) './colors';
```

Example 2:
```less
// lesshint-disable stringQuotes
@import (reference) './colors';
// lesshint-enable stringQuotes

* {
    box-sizing: border-box;
}
```

Example 3:
```less
@tablet: ~'only screen and (min-width: 481px)';
```

Example 4:
```less
.my-background-mixin {
    background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='@{width}' height='@{height}' viewBox='0 0 @{width} @{height}'%3E%3Crect width='@{width}' height='@{height}' fill='@{background}'/%3E%3Cpolyline transform='translate(@{offsetHorizontal} @{offsetVertical})' points='0 0 6 6 12 0' fill='none' stroke='%23000' stroke-linecap='round' stroke-linejoin='round' stroke-width='2'/%3E%3C/svg%3E");
}
```

Example 5:
```less
@just-text-type-input-selectors: &:not([type='file']):not([type='checkbox']):not([type='radio']):not([type='range']):not([type='hidden']);
}
```

**What did you expect to happen?**

No lesshint warnings.

**What's the actual result?**

Example 1: `stringQuotes` warning.
Example 2: `newlineAfterBlock` warning on the line of `* {` - this is probably caused by #504.
Example 3: `stringQuotes` warning.
Example 4: `stringQuotes` warning - which is unfortunate as it contradicts [Prettier's logic](https://prettier.io/docs/en/options.html#quotes), and in this case we need both types of quotes.
Example 5: `stringQuotes` warning.

**Would you be interested in submitting a PR for this issue?**

Phew, sorry, no.

**Comment by jwilsson** [comment]
Thanks for the report and for using the template!

I'm thinking a new option, `avoidEscape` (à la [stylelint](https://github.com/stylelint/stylelint/tree/master/lib/rules/string-quotes#avoidescape-truefalse-defaults-to-true)), for example 4. And bug fixes for the rest.

**Comment by jwilsson** [comment]
Fixes for this, and the option is available in `lesshint@6.1.0`.
